src/dhcpmon/src/dhcp_device.cpp

```diff
@@ -221,6 +231,88 @@ static void read_callback(int fd, short event, void *arg)
     }
 }
 
+/**
+ * @code read_callback_dual_tor(fd, event, arg);
+ *
+ * @brief callback for libevent which is called every time out in order to read queued packet capture when dual tor mode is enabled
+ *
+ * @param fd            socket to read from
+ * @param event         libevent triggered event
+ * @param arg           user provided argument for callback (interface context)
+ *
+ * @return none
+ */
+static void read_callback_dual_tor(int fd, short event, void *arg)
+{
+    dhcp_device_context_t *context = (dhcp_device_context_t*) arg;
+    ssize_t buffer_sz;
+    struct sockaddr_ll sll;
+    socklen_t slen = sizeof sll;
+
+    bool standby = false;
+
+    while ((event == EV_READ) &&
+           ((buffer_sz = recvfrom(fd, context->buffer, context->snaplen, MSG_DONTWAIT, (struct sockaddr *)&sll, &slen)) > 0)) {
+        std::string member_table = std::string("VLAN_MEMBER|") + context->intf + "|";
+        char interfaceName;
+  	    char *interface = if_indextoname(sll.sll_ifindex, &interfaceName);
+
+        std::string state;
+        std::string intf(interface);
+        mStateDbMuxTablePtr->hget(intf, "state", state);
+        if (state != "standby" && configDb.exists(member_table.append(interface))) {
+            struct ether_header *ethhdr = (struct ether_header*) context->buffer;
+            struct ip *iphdr = (struct ip*) (context->buffer + IP_START_OFFSET);
+            struct udphdr *udp = (struct udphdr*) (context->buffer + UDP_START_OFFSET);
+            uint8_t *dhcphdr = context->buffer + DHCP_START_OFFSET;
+            int dhcp_option_offset = DHCP_START_OFFSET + DHCP_OPTIONS_HEADER_SIZE;
+
+            if ((buffer_sz > UDP_START_OFFSET + sizeof(struct udphdr) + DHCP_OPTIONS_HEADER_SIZE) &&
+                (ntohs(udp->len) > DHCP_OPTIONS_HEADER_SIZE)) {
+                int dhcp_sz = ntohs(udp->len) < buffer_sz - UDP_START_OFFSET - sizeof(struct udphdr) ?
+                            ntohs(~udp->len) : buffer_sz - UDP_START_OFFSET - sizeof(struct udphdr);
```

Comment:
What will negative dhcp_sz do?

Reply:
If dhcp_sz is somehow negative, while loop below won't be executed

Reply:
Updated dhcp_sz to be 0